cli/flox/doc/flox-config.md

```diff
@@ -7,69 +7,112 @@ header: "flox User Manuals"
 
 # NAME
 
-flox-config - configure user parameters
+flox-config - view and set configuration options
 
 # SYNOPSIS
 
-flox [ `<general-options>` ] config ([(-l|\--list)] | (-r|\--reset) | --set `<key>` `<value>` | --set-number `<key>` `<number>` | --set-bool `<key>` `<bool>` | --delete `<key>`)
+```
+flox [ <general-options> ] config
+     [-l |
+      -r |
+      --set <key> <string> |
+      --set-number <key> <number> |
+      --set-bool <key> <bool> |
+      --delete=<key>]
+```
 
 # DESCRIPTION
 
-Configure and/or display user-specific parameters.
+Without any flags or when `-l` is passed, `flox config` shows all options with
+their computed value.
+
+The values are computed by reading:
+
+1. `flox` provided defaults.
+2. System settings from `/etc/flox.toml`.
+3. User customizations from `$XDG_CONFIG_DIRS`.
+4. User customizations from `$FLOX_CONFIG_HOME` if set or else `$XDG_CONFIG_HOME`.
+4. Environment variables.
+   All config options may be set by prefixing with `FLOX_` and using
+   SCREAMING_SNAKE_CASE.
+   For example, `disable_metrics` may be set with `FLOX_DISABLE_METRICS=true`.
 
-Without any flags or `-l` or `--list` shows all configurable options
-with their computed value.
+The last occurence is used as the final value.
+
+`flox config` commands that mutate configuration always write to
+`${FLOX_CONFIG_HOME:-$XDG_CONFIG_HOME}/flox/flox.toml`.
 
 ## Key Format
 
-`<key>` supports dot-separated queries for nested vaules, e.g.
+`<key>` supports dot-separated queries for nested values, for example:
 
 ```
-flox config --set 'nix.access_tokens."github.com"' "ghp_xxx"`
+flox config --set 'trusted_environments."owner/name"' trust
 ```
 
-## Config Format
+# OPTIONS
 
-All config keys can be listed with
+## Config Options
 
-```
-flox config
-```
+`-l`, `--list`
+:   List the current values of all options.
 
-The dispalyed values are resolved by reading:
+`-r`, `--reset`
+:   Reset all options to their default values without further confirmation.
 
-1. package defaults from `$PREFIX/etc/flox.toml`
-2. installation defaults from `/etc/flox.toml`
-3. user customizations from `$HOME/.config/flox/flox.toml`
-4. environment variables
+`--set <key> <value>`
+:  Set `<key> = <value>` for string values
 
-where the last occurence is used as the final value.
+`--set-number <key> <value>`
+:  Set `<key> = <value>` for number values
 
-`flox config` commands that mutate configurations always write to `$HOME/.config/flox/flox.toml`.
+`--set-bool <key> <value>`
+:  Set `<key> = <value>` for boolean values
 
-# OPTIONS
+`--delete <key>`
+:   Delete config key
 
 ```{.include}
 ./include/general-options.md
-./include/development-options.md
 ```
 
-## Config Options
+# SUPPORTED CONFIGURATION OPTIONS
+
+`config_dir`
+:   Directory where flox should load its configuration file (default:
+    `$XDG_CONFIG_HOME/flox`).
+    This option will only take effect if set with `$FLOX_CONFIG_HOME`.
+    `$FLOX_CONFIG_DIR` and `config_dir` are ignored.
```

Comment:
I think we should probably remove `config_dir` from config files and drop `$FLOX_CONFIG_HOME` in favor of `$FLOX_CONFIG_DIR` for consistency

Reply:
`{cache,data,config}_dir` all refer to `XDG_XXX_HOME`, but are exposed as `FLOX_XXX_DIR` I agree we should probably align on either `FLOX_XXX_DIR` or `FLOX_XXX_HOME`.

`config_dir` is included in the config for completeness and I'd rather leave it in the struct.
Though overriding this via anything but the env var is not supported (kind of a bootstrapping issue there).

Reply:
Regardless of whether we move it out of the struct, it's pretty buggy to use it for part of config but not other parts. I think right now it's used for e.g. global manifest but not flox.toml

Reply:
What is used by global manifest but not flox.toml?
Lost context sorry

Reply:
Didn't explain that too well. We have this weird situation where `config_dir` in a `flox.toml` impacts some things (e.g. global-manifest) but not others (the config dir itself)
```
> mkdir /tmp/1
> echo 'floxhub_url = "https://example1.com"' > /tmp/1/flox.toml
> echo 'config_dir = "/tmp/2"' >> /tmp/1/flox.toml
> FLOX_CONFIG_HOME=/tmp/1 flox config
Failed to initialize global manifest: No such file or directory (os error 2)

Please ensure that you have write permissions
to write '~/.config/flox/global-manifest.toml'.

> mkdir /tmp/2
> FLOX_CONFIG_HOME=/tmp/1 flox config
config_dir = "/tmp/2"
floxhub_url = "https://example1.com/"
```

Reply:
Ah yeah the config for should not be overridable that's going to be weird

Reply:
https://github.com/flox/flox/issues/940